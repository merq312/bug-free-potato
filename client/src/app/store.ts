import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit"
import { logger } from "redux-logger"
import counterReducer from "../features/counter/counterSlice"
import messageReducer from "../features/messages/messageSlice"
import userReducer from "../features/user/userSlice"
import { createSocketMiddleware } from "../features/middleware/socket.middleware"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    messages: messageReducer,
    user: userReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(createSocketMiddleware(), logger),
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
