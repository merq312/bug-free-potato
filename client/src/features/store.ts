import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit"
import { logger } from "redux-logger"
import counterReducer from "./counter/counterSlice"
import messageReducer from "./message/messageSlice"
import userReducer from "./user/userSlice"
import roomReducer from "./room/roomSlice"
import { createSocketMiddleware } from "./middleware/socket.middleware"

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    messages: messageReducer,
    user: userReducer,
    rooms: roomReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(createSocketMiddleware(), logger),
})

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>
