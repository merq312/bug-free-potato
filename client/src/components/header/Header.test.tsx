import {render} from "@testing-library/react";
import HeaderComponent from "./Header.component";
import {Provider} from "react-redux";
import {store} from "../../features/store";

describe("Header Component", () => {
  test("renders correctly", () => {
    const {asFragment} = render(
      <Provider store={store}>
        <HeaderComponent/>
      </Provider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
