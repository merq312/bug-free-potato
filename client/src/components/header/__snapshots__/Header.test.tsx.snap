// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="bg-gray-700 flex items-center justify-between w-screen h-16 border-b-2 border-gray-200"
  >
    <h1
      class="text-gray-200 text-xl sm:text-2xl font-bold m-2"
    >
      Some Message App
    </h1>
    <p
      class="text-gray-200 px-4"
    >
      Loading...
    </p>
  </header>
</DocumentFragment>
`;
